app/scripts/controllers/network/network-controller.js

```diff
@@ -476,4 +498,67 @@ export default class NetworkController extends EventEmitter {
     this._provider = provider;
     this._blockTracker = blockTracker;
   }
+
+  /**
+   * Network Configuration management functions
+   */
+
+  /**
+   * Adds a network configuration if the rpcUrl is not already present on an
+   * existing network configuration. Otherwise updates the entry with the matching rpcUrl.
+   *
+   * @param config - The network configuration.
```

Comment:
Nit: Missing JSDoc types in this comment block, and the following one.

Reply:
added [here](https://github.com/MetaMask/metamask-extension/pull/17421/commits/930d47f53d5c35b5b6ef1ffa1c711dbce1bef125)